src/main/kotlin/org/jitsi/nlj/rtp/TransportCcEngine.kt

```diff
@@ -100,37 +104,47 @@ class TransportCcEngine(
             localReferenceTime = now
         }
 
-        // We have to remember the oldest known sequence number here, as we
-        // remove from sentPacketDetails inside this loop
-        val oldestKnownSeqNum = synchronized(sentPacketsSyncRoot) { sentPacketDetails.oldestEntry() }
         for (packetReport in tccPacket) {
             val tccSeqNum = packetReport.seqNum
             val packetDetail = synchronized(sentPacketsSyncRoot) {
-                sentPacketDetails.remove(tccSeqNum)
+                sentPacketDetails.get(tccSeqNum)
             }
 
             if (packetDetail == null) {
                 if (packetReport is ReceivedPacketReport) {
+                    currArrivalTimestamp += packetReport.deltaDuration
                     missingPacketDetailSeqNums.add(tccSeqNum)
                 }
                 continue
             }
 
             when (packetReport) {
-                is UnreceivedPacketReport ->
-                    bandwidthEstimator.processPacketLoss(now, packetDetail.packetSendTime, tccSeqNum)
-
+                is UnreceivedPacketReport -> {
+                    if (packetDetail.state == PacketDetailState.unreported) {
+                        bandwidthEstimator.processPacketLoss(now, packetDetail.packetSendTime, tccSeqNum)
+                        packetDetail.state = PacketDetailState.reportedLost
```

Comment:
I wonder if we should use a different name here other than `reportedLost`, since tcc doesn't technically ever denote that.